source/client-side-encryption/tests/README.rst

```diff
@@ -596,6 +606,99 @@ Data keys created with AWS KMS may specify a custom endpoint to contact (instead
 
    Expect this to fail with an exception with a message containing the string: "parse error"
 
+Azure and GCP custom endpoints
+``````````````````````````````
+
+Test specifying custom endpoints for authenticating Azure and GCP and for a custom GCP KMS endpoint.
+
+1. Create a ``ClientEncryption`` object (referred to as ``client_encryption``).
+
+   Configure with KMS providers as follows:
+
+   .. code:: javascript
+
+      {
+         "azure": {
+            "tenantId": <set from environment>,
+            "clientId": <set from environment>,
+            "clientSecret": <set from environment>,
+            "identityPlatformEndpoint": "login.microsoftonline.com:443"
+         },
+            "gcp": {
+            "email": <set from environment>,
+            "privateKey": <set from environment>,
+            "endpoint": "oauth2.googleapis.com:443"
+         }
+      }
+
+   Create a ``ClientEncryption`` object (referred to as ``client_encryption_invalid``).
+
+   Create a new ``ClientEncryption`` object (referred to as ``client_encryption_invalid``).
+
+   Configure with KMS providers as follows:
+
+   .. code:: javascript
+
+      {
+         "azure": {
+            "tenantId": <set from environment>,
+            "clientId": <set from environment>,
+            "clientSecret": <set from environment>,
+            "identityPlatformEndpoint": "example.com:443"
+         },
+            "gcp": {
+            "email": <set from environment>,
+            "privateKey": <set from environment>,
+            "endpoint": "example.com:443"
+         }
+      }
+
+   Configure both ``ClientEncryption`` objects with ``keyVaultNamespace`` set to ``keyvault.datakeys``, and a default MongoClient as the ``keyVaultClient``.
+
+2. Call `client_encryption.createDataKey()` with "azure" as the provider and the following masterKey:
+
+   .. code:: javascript
+
+      {
+        "keyVaultEndpoint": "key-vault-kevinalbs.vault.azure.net",
```

Comment:
This, and the other masterKey info, will be changed when we switch from my personal accounts to company managed Azure/GCP accounts.

Reply:
Updated.